libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md

```diff
@@ -0,0 +1,131 @@
+# 3DR (mRo) Control Zero H7 OEM Flight Controller revision G
+
+The Control Zero H7 OEM revision G is a flight controller produced by [3DR (mRo)](https://store.3dr.com/control-zero-h7-oem-g/).
+
+![3DR Control Zero H7 OEM rev G - Top](https://vddwxegfxugwzpfnrrlp.supabase.co/storage/v1/object/public/Website-CDN/autopilot-img/CZOEM_revG_front.JPG)
+![3DR Control Zero H7 OEM rev G - Bottom](https://vddwxegfxugwzpfnrrlp.supabase.co/storage/v1/object/public/Website-CDN/autopilot-img/CZOEM_revG_back.JPG?t=2024-03-08T20%3A18%3A49.140Z)
+![3DR Control Zero H7 OEM rev G - Top w/ case](https://vddwxegfxugwzpfnrrlp.supabase.co/storage/v1/object/public/Website-CDN/autopilot-img/CZOEM_revG_case_front.JPG?t=2024-03-08T20%3A18%3A57.128Z)
+![3DR Control Zero H7 OEM rev G - Bottomi w/ case](https://vddwxegfxugwzpfnrrlp.supabase.co/storage/v1/object/public/Website-CDN/autopilot-img/CZOEM_revG_case.jpg?t=2024-03-08T20%3A18%3A52.750Z)
+
+## Features
+
+	Processor
+		STM32H743IIK6 32-bit Processor
+	Sensors
+		BMI088 6DOF
+		ICM20602 6DOF
+		ICM20948 9DOF
+		DPS368 Baro
+	Power
+		External Power Supply
+		Logic level at 3.3V
+	Interfaces
+		Bottom Connectors: 36pin front and 40pin back Samtec FTM-118-02-x and FTM-120-02-x
+		8x PWM / IO - DMA capable
+		1x RC Input 
+		5x UARTs (2x with hardware flow control)
+		2x CAN
+		1x SPI
+		3x I2C
+		SWD via TC2030 header
+		SDCARD Socket
+	Memory
+		FRAM (256KB)
+	Miscellaneous
+		Onboard 3 color LED
+		Buzzer
+		Safety Button
+
+
+###  Uncased Weight and Dimensions
+
+ Weight: 3.66g (13.oz)
+ Width:  20mm (.79in)
+ Length: 34mm (1.34in)
+
+ *Case sold separately*
+
+## Changelog
+
+- M10059C - Initial Release
+- M10059G adds external power supply and TCXO.
+
+## Pinout
+
+![Control Zero H7 OEM revision G pinout](czpinout_revG.png)
+
+## UART Mapping
+
+- SERIAL0 -> USB1, for GCS connection
+
+- SERIAL1 -> USART2 (TELEM 1) DMA Enabled
+
+- SERIAL2 -> USART3 (TELEM 2) DMA Enabled
+
+- SERIAL3 -> UART4 (GPS) DMA Enabled
+
+- SERIAL4 -> UART8 (FRSKY TELEM) DMA Enabled
+
+- SERIAL5 -> UART7 (DEBUG) DMA Enabled
+
+- SERIAL6 -> USB2, MAVLink interface
+
+*UART6 is available upon request, more info at info@3dr.com* 
+
+## RC Input
+
+RC input is configured on the RC_IN pin. These are the supported RC input protocols: 
+
+Spektrum DSM / DSM2 / DSM-X® Satellite compatible input and binding. 
+Futaba S.BUS® & S.BUS2® compatible input. 
+FrSky Telemetry port output. 
```

Comment:
this not an rc protocol...if yu mean FPort, the it requires a half duplex serial option on the uart port and connection to the tx oin or rx/tx swap option also